wordpress-mobile/WordPress-Android
# Fixes: Logout needs to be pressed twice

Fixes 
#15976 

Changes 
When the logout button was pressed, the following code was getting processed 
```
if (!BuildConfig.IS_JETPACK_APP || mSiteStore.hasSite()) {
// delete wpcom and jetpack sites
mDispatcher.dispatch(SiteActionBuilder.newRemoveWpcomAndJetpackSitesAction());
}
```
This would emit a `SiteAction.REMOVE_WPCOM_AND_JETPACK_SITES` and this, in turn, would emit an `OnSiteRemoved`, which is subscribed in `WPMainActivity`. `WPMainActivity` will show a site picker or show a sign-in activity screen depending on whether the user has any other site. Since there was a check if the app for Jetpack in the code, this was happening on the Jetpack app for a wp.com account having no sites. This code was not getting called in a Jetpack pack for a wp.com account with no sites, but in the WordPress app, this would be called for the same account. 


https://user-images.githubusercontent.com/17463767/156581928-9d8f039a-3ec0-4058-82a9-767225af9b31.mp4

----

**I have tested the following scenarios** 

WP.com account in WordPress app 
- Logging out from the WordPress app for a `wp.com` account with sites works as expected 
- Logging out from the WordPress app for a `wp.com` account having no sites works as expected 

WP.com account in Jetpack app
- Logging out from the Jetpack app for a `wp.com` account having sites works as expected 
- Logging out from the Jetpack app for a `wp.com` account having no sites works as expected 

---

To test:
1. Login to Jetpack app with a wpcom account having no sites.
2. Go to My Site tab and log out from the Me profile settings.
3. Notice that on pressing Logout button, you are logged out. 

## Regression Notes
1. Potential unintended areas of impact
Logging out from WordPress and Jetpack for accounts having sites/no sites doesn't work as expected

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests 


4. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
